app/services/convert_deprecated_csharp_parameters_service.rb

```diff
@@ -0,0 +1,48 @@
+class ConvertDeprecatedCsharpParametersService
+  def call(parameters:)
```

Comment:
Trying to infer parameter types actually ends up creating more complexity (because for arrays you have to exclude other types) and some types can be ambiguous which makes it more prone to edge cases.